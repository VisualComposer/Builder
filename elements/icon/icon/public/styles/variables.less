// icon sizes
@vce-icon-font-size-tiny: 18px;
@vce-icon-font-size-small: 24px;
@vce-icon-font-size-medium: 32px;
@vce-icon-font-size-large: 48px;
