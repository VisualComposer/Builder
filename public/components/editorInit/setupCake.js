import vcCake from 'vc-cake'
export const setupCake = () => {
  vcCake.env('platform', 'wordpress').start(() => {
    vcCake.env('editor', 'frontend')
    require('../../editor/stores/fieldOptionsStorage')
    require('../../editor/stores/events/eventsStorage')
    require('../../editor/stores/elements/elementsStorage')
    require('../../editor/stores/assets/assetsStorage')
    require('../../editor/stores/shortcodesAssets/storage')
    require('../../editor/stores/cacheStorage')
    require('../../editor/stores/migrationStorage')
    require('../../editor/stores/workspaceStorage')
    require('../../editor/stores/hub/hubElementsStorage')
    require('../../editor/stores/hub/hubTemplatesStorage')
    require('../../editor/stores/hub/hubAddonsStorage')
    require('../../editor/stores/sharedAssets/storage')
    require('../../editor/stores/history/historyStorage')
    require('../../editor/stores/settingsStorage')
    require('../../editor/stores/attributes/attributesStorage')
    require('../../editor/stores/notifications/storage')
    require('../../editor/stores/wordpressData/wordpressDataStorage')
    require('../../editor/stores/elements/elementSettings')
    require('../../editor/stores/popup/storage')
    // require('./editor/stores/elementsLoader/elementsLoaderStorage')
    const hubElementsStorage = vcCake.getStorage('hubElements')
    hubElementsStorage.trigger('start')
    const hubTemplatesStorage = vcCake.getStorage('hubTemplates')
    hubTemplatesStorage.trigger('start')
    const hubAddonsStorage = vcCake.getStorage('hubAddons')
    hubAddonsStorage.trigger('start')
    const sharedAssetsStorage = vcCake.getStorage('sharedAssets')
    sharedAssetsStorage.trigger('start')
    const settingsStorage = vcCake.getStorage('settings')
    settingsStorage.trigger('start')
    const attributesStorage = vcCake.getStorage('attributes')
    attributesStorage.trigger('start')
    require('../../config/wp-modules')
  })
}
