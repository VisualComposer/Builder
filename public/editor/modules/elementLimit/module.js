import { getStorage, getService } from 'vc-cake'
import store from 'public/editor/stores/store'
import { notificationAdded } from 'public/editor/stores/notifications/slice'

const elementsStorage = getStorage('elements')
const documentManager = getService('document')
const cook = getService('cook')
const dataManager = getService('dataManager')

const localizations = dataManager.get('localizations')
const limitTexts = {
  elementLimit1: localizations.onlyOneElementCanBeAddedToPage || 'Only one %element element can be added to the page.',
  elementLimit2: localizations.onlyTwoElementsCanBeAddedToPage || 'Only two %element elements can be added to the page.',
  elementLimit3: localizations.onlyThreeElementsCanBeAddedToPage || 'Only three %element elements can be added to the page.',
  elementLimit4: localizations.onlyFourElementsCanBeAddedToPage || 'Only four %element elements can be added to the page.',
  elementLimit5: localizations.onlyFiveElementsCanBeAddedToPage || 'Only five %element elements can be added to the page.',
  elementLimitDefault: localizations.elementLimitDefaultText || 'Only %count %element elements can be added to the page.'
}

const triggerNotification = (type, elementName, limit) => {
  let limitText = limitTexts.elementLimitDefault
  if (limit < 6) {
    limitText = limitTexts[`elementLimit${limit}`]
  } else {
    limitText = limitText.replace('%count', limit)
  }
  limitText = limitText.replace('%element', elementName)

  store.dispatch(notificationAdded({
    type: type,
    text: limitText,
    time: 5000,
    showCloseButton: true
  }))
}

const getElementExceededLimitStatus = (element) => {
  const limitData = {}
  if (Object.prototype.hasOwnProperty.call(element, 'metaElementLimit')) {
    const limit = parseInt(element.metaElementLimit)
    const limitedElements = documentManager.getByTag(element.tag) || {}
    if (limit > 0 && Object.keys(limitedElements).length >= limit) {
      limitData.hasExceeded = true
      limitData.limit = limit
    }
  }
  return limitData
}

elementsStorage.registerAction('beforeAdd', (element) => {
  const elementLimitData = getElementExceededLimitStatus(element)
  if (elementLimitData.hasExceeded) {
    const cookElement = cook.get(element)
    triggerNotification('error', cookElement.get('name'), elementLimitData.limit)
    return true
  }
  return false
})

elementsStorage.registerAction('beforeClone', (elementID) => {
  const element = documentManager.get(elementID)
  const elementLimitData = getElementExceededLimitStatus(element)
  if (elementLimitData.hasExceeded) {
    const cookElement = cook.get(element)
    triggerNotification('error', cookElement.get('name'), elementLimitData.limit)
    return true
  }
  return false
})
