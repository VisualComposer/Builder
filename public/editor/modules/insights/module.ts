import { getStorage, env, add } from 'vc-cake'
import { debounce } from 'lodash'
import InsightsChecks from './lib/InsightsChecks'
import store from 'public/editor/stores/store'
import { insightsReset } from 'public/editor/stores/insights/slice'

const historyStorage = getStorage('history')
const settingsStorage = getStorage('settings')
const workspaceStorage = getStorage('workspace')
const elementsStorage = getStorage('elements')

add('insights', () => {
  if (typeof (window as any).vcvRebuildPostSave !== 'undefined') {
    return
  }
  if (env('VCV_FT_INSIGHTS')) {
    const insightsChecksInstance = new InsightsChecks()
    const runChecksCallback = debounce(() => {
      const iframeDomNode = document.querySelector<HTMLIFrameElement>('.vcv-layout-iframe')
      if (!iframeDomNode || !iframeDomNode.contentWindow) {
        return // editor reload
      }
      // clear previous <Insights>
      store.dispatch(insightsReset())
      insightsChecksInstance.isImagesSizeLarge = false

      // Do all checks
      const isEmptyContent: boolean = insightsChecksInstance.checkForEmptyContent()
      if (!isEmptyContent) {
        insightsChecksInstance.checkForHeadings()
        insightsChecksInstance.checkForAlt()
        insightsChecksInstance.checkNameForLinks()
        insightsChecksInstance.checkForImagesSize()
        insightsChecksInstance.checkParagraphsLength()
        insightsChecksInstance.checkTitleLength()
        insightsChecksInstance.checkNoIndex()
        insightsChecksInstance.checkPostContentLength()
        insightsChecksInstance.checkForGA()
        insightsChecksInstance.checkLinks()
        insightsChecksInstance.contrast()
      }
    }, 5000)
    historyStorage.on('init add undo redo', runChecksCallback)
    settingsStorage.state('pageTitleDisabled').onChange(runChecksCallback)
    settingsStorage.state('pageTitle').onChange(runChecksCallback)
    workspaceStorage.state('iframe').onChange(({ type }: { type: string }) => {
      if (type === 'loaded') {
        runChecksCallback()
      }
    })
    elementsStorage.on('elementsRenderDone', runChecksCallback)
  }
})
