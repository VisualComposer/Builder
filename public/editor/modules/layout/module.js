/* global VCV_PLUGIN_UPDATE */
import vcCake from 'vc-cake'
import React from 'react'
import ReactDOM from 'react-dom'
import Editor from './lib/editor'
import DndManager from './lib/dndManager'
import ControlsManager from './lib/controlsIframe/controlsManager'
import RightClickMenu from 'public/components/rightClickMenu/rightClickMenu'
import MobileControlsManager from './lib/controlsIframe/mobileControlsManager'
import MobileDetect from 'mobile-detect'
import OopsScreen from 'public/components/account/oopsScreen'
import Notifications from 'public/components/notifications/notifications'
import Popup from 'public/components/popup/popupContainer'
import FullPagePopupContainer from 'public/components/popup/fullPagePopupContainer'
import Helpers from 'public/components/helpers/helpers'

const notificationsStorage = vcCake.getStorage('notifications')
const Utils = vcCake.getService('utils')
const workspaceStorage = vcCake.getStorage('workspace')
const workspaceSettings = workspaceStorage.state('settings')
const workspaceIFrame = workspaceStorage.state('iframe')
const elementsStorage = vcCake.getStorage('elements')
const assetsStorage = vcCake.getStorage('assets')
const dataManager = vcCake.getService('dataManager')

vcCake.add('contentLayout', (api) => {
  const iframeContent = document.getElementById('vcv-layout-iframe-content')
  const dnd = new DndManager(api)
  const controls = new ControlsManager(api)
  const rightClick = new RightClickMenu()

  const localizations = dataManager.get('localizations')
  if (Utils.isRTL()) {
    document.body && document.body.classList.add('rtl')
  }

  // Start notifications and editor popups
  const layoutOverlay = document.querySelector('.vcv-layout-overlay')
  if (layoutOverlay) {
    ReactDOM.render(
      <>
        <Notifications />
        <Popup />
        <FullPagePopupContainer />
        {dataManager.get('showInitialHelpers') && <Helpers />}
      </>,
      layoutOverlay
    )
  }

  const renderLayout = (reload = false) => {
    workspaceIFrame.ignoreChange(reloadLayout)
    workspaceIFrame.set(false)
    const iframe = document.getElementById('vcv-editor-iframe')
    const iframeWindow = iframe ? iframe.contentWindow : null
    const domContainer = iframeWindow ? iframeWindow.document.getElementById('vcv-editor') : null
    if (domContainer) {
      ReactDOM.render(
        <Editor api={api} />,
        domContainer
      )

      !reload && dnd.init()
      !reload && rightClick.init()

      workspaceIFrame.onChange(reloadLayout)

      const pluginUpdate = typeof VCV_PLUGIN_UPDATE === 'function' ? VCV_PLUGIN_UPDATE() : false
      pluginUpdate && notificationsStorage.trigger('add', {
        position: 'top',
        transparent: false,
        showCloseButton: true,
        rounded: false,
        type: 'warning',
        text: localizations.newPluginVersionIsAvailable || 'There is a new version of Visual Composer Website Builder available',
        html: true,
        cookie: {
          name: 'vcv-update-notice',
          expireInDays: 1
        },
        time: 5000
      })

      const mobileDetect = new MobileDetect(window.navigator.userAgent)
      if (mobileDetect.mobile() && (mobileDetect.tablet() || mobileDetect.phone())) {
        const mobileControls = new MobileControlsManager(api)
        mobileControls.init()

        notificationsStorage.trigger('add', {
          position: 'bottom',
          transparent: true,
          rounded: true,
          text: localizations.mobileTooltipText || 'Double-tap on an element to open the edit window. Tap and hold to initiate drag and drop.',
          cookie: 'vcv-mobile-tooltip',
          time: 10000
        })
        return
      }

      reload ? controls.updateIframeVariables() : controls.init()
      if (vcCake.env('VCV_JS_THEME_LAYOUTS')) {
        iframeWindow.document.querySelectorAll('[data-vcv-layout-zone]').forEach((zone) => {
          const zoneButton = zone.querySelector('[data-vcv-action="settings"]')
          zoneButton && zoneButton.addEventListener('click', () => {
            workspaceStorage.state('content').set('settings')
            workspaceSettings.set({ action: 'settings' })
          })
        })
      }
    } else {
      const postUpdateWrapper = document.querySelector('#vcv-posts-update-wrapper')
      let wrapper = document.body

      if (postUpdateWrapper) {
        const postUpdateIframe = postUpdateWrapper.querySelector('.vcv-layout-iframe')
        wrapper = postUpdateIframe && postUpdateIframe.contentWindow.document.body
      }

      wrapper.innerHTML = '<div id=\'vcv-oops-screen-container\'></div>'
      const oopsContainer = document.getElementById('vcv-oops-screen-container')
      if (oopsContainer) {
        ReactDOM.render(
          <div className='vcv-screen-section'>
            <OopsScreen errorName={dataManager.get('frontEndError')} />
          </div>,
          oopsContainer
        )
      }
    }
  }

  const removeLoadingScreen = () => {
    let loadingOverlays = iframeContent.querySelectorAll('.vcv-loading-overlay')
    loadingOverlays = [].slice.call(loadingOverlays)
    loadingOverlays.forEach((loadingOverlay) => {
      loadingOverlay.remove()
    })
  }

  const createLoadingScreen = () => {
    removeLoadingScreen()
    const loadingOverlay = document.createElement('div')
    loadingOverlay.classList.add('vcv-loading-overlay')
    loadingOverlay.innerHTML = `<div class='vcv-loading-overlay-inner'>
        <div class='vcv-loading-dots-container'>
          <div class='vcv-loading-dot vcv-loading-dot-1'></div>
          <div class='vcv-loading-dot vcv-loading-dot-2'></div>
        </div>
      </div>`
    const startBlank = iframeContent.querySelector('.vcv-start-blank-container')
    if (startBlank) {
      iframeContent.insertBefore(loadingOverlay, startBlank)
    } else {
      iframeContent.appendChild(loadingOverlay)
    }
  }

  const reloadLayout = ({ type, template, header, sidebar, footer }) => {
    if (type === 'reload') {
      createLoadingScreen()
      const iframe = window.document.getElementById('vcv-editor-iframe')
      const domContainer = iframe.contentDocument.getElementById('vcv-editor')
      if (domContainer) {
        ReactDOM.unmountComponentAtNode(domContainer)
      }
      iframe.onload = () => {
        assetsStorage.trigger('reset')
        const data = vcCake.getService('document').all()
        const visibleElements = Utils.getVisibleElements(data)

        const setIframeLoaded = () => {
          workspaceIFrame.set({ type: 'layoutLoaded' })
          elementsStorage.off('elementsRenderDone', setIframeLoaded)
        }

        elementsStorage.on('elementsRenderDone', setIframeLoaded)

        workspaceIFrame.set({ type: 'loaded' })
        window.setTimeout(() => {
          elementsStorage.trigger('updateAll', data)
          assetsStorage.trigger('updateAllElements', visibleElements)
        }, 1)
      }
      const url = iframe.src.split('?')
      let params = url[1].split('&')
      params = params.reduce((arr, item) => {
        let write = true
        if (item.indexOf('vcv-template') >= 0) {
          write = false
        }
        if (item.indexOf('vcv-nonce') >= 0) {
          write = false
        }
        if (vcCake.env('VCV_JS_THEME_LAYOUTS')) {
          if (
            item.indexOf('vcv-header') >= 0 ||
            item.indexOf('vcv-sidebar') >= 0 ||
            item.indexOf('vcv-footer') >= 0
          ) {
            write = false
          }
        }
        write && arr.push(item)
        return arr
      }, [])
      params.push(`vcv-nonce=${dataManager.get('pageEditableNonce')}`)
      if (template) {
        params.push(`vcv-template=${template.value}`)
        params.push(`vcv-template-type=${template.type}`)
        params.push(`vcv-template-stretched=${template.stretchedContent ? 1 : 0}`)

        let hasHeader = false
        let hasSidebar = false
        let hasFooter = false
        const layouts = dataManager.get('pageTemplatesLayouts')
        const currentLayoutType = layouts && layouts.find(item => item.type === template.type)
        if (currentLayoutType && currentLayoutType.values) {
          const currentTemplate = currentLayoutType.values.find(item => item.value === template.value)
          if (currentTemplate) {
            hasHeader = currentTemplate.header
            hasSidebar = currentTemplate.sidebar
            hasFooter = currentTemplate.footer
          }
        }
        if (template.type === 'theme') {
          hasHeader = true
          hasFooter = true
        }
        if (vcCake.env('VCV_JS_THEME_LAYOUTS')) {
          if (hasHeader && header) {
            params.push(`vcv-header=${header}`)
          }
          if (hasSidebar && sidebar) {
            params.push(`vcv-sidebar=${sidebar}`)
          }
          if (hasFooter && footer) {
            params.push(`vcv-footer=${footer}`)
          }
        }
      }
      url[1] = params.join('&')
      iframe.src = url.join('?')
    } else if (type === 'loaded') {
      renderLayout(true)
    } else if (type === 'layoutLoaded') {
      removeLoadingScreen()
    }
  }

  renderLayout()

  assetsStorage.state('jobs').onChange((data) => {
    const documentElements = vcCake.getService('document').all()
    if (documentElements) {
      const visibleJobs = data.elements.filter(element => !element.hidden)
      const visibleElements = Utils.getVisibleElements(documentElements)
      const documentIds = Object.keys(visibleElements)
      if (documentIds.length === visibleJobs.length) {
        const jobsInprogress = data.elements.find(element => element.jobs)
        if (jobsInprogress) {
          return
        }
        elementsStorage.trigger('elementsCssBuildDone', data)
      }
    }
  })

  elementsStorage.on('elementsCssBuildDone', () => {
    // need wait until latest element will be rendered
    let timer = null
    const renderDone = () => {
      elementsStorage.trigger('elementsRenderDone')
    }
    api.on('element:didUpdate', () => {
      window.clearTimeout(timer)
      timer = null
      timer = window.setTimeout(renderDone, 150)
    })

    timer = window.setTimeout(renderDone, 200)
  })
})
