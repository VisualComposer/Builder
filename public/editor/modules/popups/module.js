import { add, getStorage, getService } from 'vc-cake'
import { getPopupDataFromElement } from 'public/tools/popup'
import store from 'public/editor/stores/store'
import { popupAddInProgressSet } from 'public/editor/stores/popup/slice'

const elementsStorage = getStorage('elements')
const dataProcessor = getService('dataProcessor')
const cook = getService('cook')

const addPopupHtml = (id) => {
  store.dispatch(popupAddInProgressSet(true))
  const contentWindow = window.document.querySelector('.vcv-layout-iframe').contentWindow
  const documentBody = contentWindow.document.body
  const domId = `vcv-popup-${id}`

  if (documentBody.querySelector(`#${domId}`)) {
    store.dispatch(popupAddInProgressSet(false))
    return
  }

  dataProcessor.appAdminServerRequest({
    'vcv-action': 'popupBuilder:getData:adminNonce',
    'vcv-source-id': id
  }).then((requestData) => {
    if (requestData && typeof requestData === 'string') {
      const popupContainer = contentWindow.document.createElement('div')
      popupContainer.id = domId
      popupContainer.className = 'vcv-popup-container'
      popupContainer.innerHTML = requestData
      documentBody.appendChild(popupContainer)
    }
    store.dispatch(popupAddInProgressSet(false))
  }, (error) => {
    store.dispatch(popupAddInProgressSet(false))
    console.warn('Failed to get the popup HTML', error)
  })
}

add('popups', () => {
  elementsStorage.on('add', (elementData) => {
    const cookElement = cook.get(elementData)
    const ids = getPopupDataFromElement(cookElement)
    ids.forEach((id) => {
      addPopupHtml(id)
    })
  })

  elementsStorage.on('update', (id) => {
    const cookElement = cook.getById(id)
    if (!cookElement) {
      return
    }
    const ids = getPopupDataFromElement(cookElement)
    ids.forEach((id) => {
      addPopupHtml(id)
    })
  })
})
