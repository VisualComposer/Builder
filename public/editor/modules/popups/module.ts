import { add, getStorage, getService } from 'vc-cake'
import { getPopupDataFromElement } from 'public/tools/popup'
import store from 'public/editor/stores/store'
import { popupAddInProgressSet } from 'public/editor/stores/popup/slice'
import { ElementData } from 'public/global'

const elementsStorage = getStorage('elements')
const dataProcessor = getService('dataProcessor')
const cook = getService('cook')

const addPopupHtml = (id:string) => {
  store.dispatch(popupAddInProgressSet(true))
  const iframe = window.document.querySelector('.vcv-layout-iframe') as HTMLIFrameElement
  const contentWindow = iframe?.contentWindow
  const documentBody = contentWindow?.document.body
  const domId = `vcv-popup-${id}`

  if (documentBody?.querySelector(`#${domId}`)) {
    setTimeout(() => {
      store.dispatch(popupAddInProgressSet(false))
    }, 0)
    return
  }

  dataProcessor.appAdminServerRequest({
    'vcv-action': 'popupBuilder:getData:adminNonce',
    'vcv-source-id': id
  }).then((requestData:string|void) => {
    if (requestData && typeof requestData === 'string') {
      const popupContainer = contentWindow?.document.createElement('div')
      if (popupContainer) {
        popupContainer.id = domId
        popupContainer.className = 'vcv-popup-container'
        popupContainer.innerHTML = requestData
        documentBody?.appendChild(popupContainer)
      }
    }
    store.dispatch(popupAddInProgressSet(false))
  }, (error:string) => {
    store.dispatch(popupAddInProgressSet(false))
    console.warn('Failed to get the popup HTML', error)
  })
}

add('popups', () => {
  elementsStorage.on('add', (elementData:ElementData) => {
    const cookElement = cook.get(elementData)
    const ids = getPopupDataFromElement(cookElement)
    ids.forEach((id) => {
      addPopupHtml(id)
    })
  })

  elementsStorage.on('update', (id:string) => {
    const cookElement = cook.getById(id)
    if (!cookElement) {
      return
    }
    const ids = getPopupDataFromElement(cookElement)
    ids.forEach((id) => {
      addPopupHtml(id)
    })
  })
})
