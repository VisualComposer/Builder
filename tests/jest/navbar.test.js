/* global describe, test, expect */
import vcCake from 'vc-cake'
import '../../public/variables'
// Services & Storages
import '../../public/editor/services/utils/service.js'
import '../../public/editor/services/dataManager/service.js'
import '../../public/editor/services/document/service.js'
import '../../public/editor/services/hubElements/service.js'
import '../../public/editor/services/cook/service.js'
import '../../public/editor/services/modernAssetsStorage/service.js'
import '../../public/editor/services/api/service.js'
import '../../public/editor/services/wordpress-post-data/service'
import '../../public/config/wp-attributes'
import '../../public/editor/stores/elements/elementsStorage'
import '../../public/editor/stores/elements/elementSettings'
import '../../public/editor/modules/elementLimit/module'
import React from 'react'
import renderer from 'react-test-renderer'
// import {cleanup, fireEvent, render} from '@testing-library/react'
import { setupCake } from '../../public/components/editorInit/setupCake'
import NavbarContainer from '../../public/components/navbar/navbarContainer'
// import Editor from '../../public/editor/modules/layout/lib/editor'

describe('Tests editor navbar', () => {
  // const navbarContainer = render(
  //   <NavbarContainer wrapperRef={(navbar) => { this.navbar = navbar }} getNavbarPosition={() => { return true}} />
  // )
  test('Navbar render', () => {
    // setupCake()
    vcCake.env('platform', 'wordpress').start(()=>{})
    vcCake.add('contentLayout', (api) => {
      let navbarRef
      const navbar = renderer.create(
        <NavbarContainer getNavbarPosition={() => {}} wrapperRef={(navbar) => { navbarRef = navbar }} />
      )
      let tree = navbar.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})