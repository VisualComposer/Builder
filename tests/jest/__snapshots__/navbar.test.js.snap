// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests editor navbar Navbar render 1`] = `
<div
  className="vcv-layout-bar-header vcv-layout-bar-header--loading"
  id="vcv-editor-header"
>
  <div
    id="vc-navbar-container"
  >
    
    <div
      className="vcv-ui-navbar-container vcv-ui-navbar-is-detached"
    >
      <nav
        className="vcv-ui-navbar vcv-ui-navbar-hide-labels"
      >
        <div
          className="vcv-ui-navbar-drag-handler vcv-ui-drag-handler"
          onMouseDown={[Function]}
        >
          <i
            className="vcv-ui-drag-handler-icon vcv-ui-icon vcv-ui-icon-drag-dots"
          />
        </div>
        <a
          className="vcv-ui-navbar-logo"
          href="undefined&vcv-ref=editor"
          rel="noopener noreferrer"
          target="_blank"
          title="Visual Composer Website Builder"
        >
          <span
            className="vcv-ui-navbar-logo-title"
          >
            Visual Composer Website Builder
          </span>
        </a>
        <span
          className="vcv-ui-navbar-control"
          data-vcv-guide-helper="plus-control"
          onClick={[Function]}
          title="Add Content"
        >
          <span
            className="vcv-ui-navbar-control-content"
          >
            <i
              className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-add"
            />
            <span>
              Add Content
            </span>
          </span>
        </span>
        <span
          className="vcv-ui-navbar-control"
          onClick={[Function]}
          title="Tree View"
        >
          <span
            className="vcv-ui-navbar-control-content"
          >
            <i
              className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-layers"
            />
            <span>
              Tree View
            </span>
          </span>
        </span>
        <div
          className="vcv-ui-navbar-controls-group vcv-ui-navbar-sandwich--stop-close"
        >
          <span
            className="vcv-ui-navbar-control"
            disabled={true}
            onClick={[Function]}
            title="Undo"
          >
            <span
              className="vcv-ui-navbar-control-content"
            >
              <i
                className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-undo"
              />
              <span>
                Undo
              </span>
            </span>
          </span>
          <span
            className="vcv-ui-navbar-control"
            disabled={true}
            onClick={[Function]}
            title="Redo"
          >
            <span
              className="vcv-ui-navbar-control-content"
            >
              <i
                className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-redo"
              />
              <span>
                Redo
              </span>
            </span>
          </span>
        </div>
        <dl
          className="vcv-ui-navbar-dropdown vcv-ui-navbar-dropdown-linear vcv-ui-pull-end"
          data-vcv-guide-helper="layout-control"
          tabIndex="0"
        >
          <dt
            className="vcv-ui-navbar-dropdown-trigger vcv-ui-navbar-control"
            title="Responsive View"
          >
            <span
              className="vcv-ui-navbar-control-content"
            >
              <i
                className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-multiple-devices"
              />
              <span>
                Responsive View
              </span>
            </span>
          </dt>
          <dd
            className="vcv-ui-navbar-dropdown-content"
          >
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
              title="Responsive View"
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-multiple-devices"
                />
                <span>
                  Responsive View
                </span>
              </span>
            </span>
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
              title="Desktop"
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-desktop"
                />
                <span>
                  Desktop
                </span>
              </span>
            </span>
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
              title="Tablet Landscape"
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-tablet-landscape"
                />
                <span>
                  Tablet Landscape
                </span>
              </span>
            </span>
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
              title="Tablet Portrait"
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-tablet-portrait"
                />
                <span>
                  Tablet Portrait
                </span>
              </span>
            </span>
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
              title="Mobile Landscape"
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-mobile-landscape"
                />
                <span>
                  Mobile Landscape
                </span>
              </span>
            </span>
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
              title="Mobile Portrait"
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-mobile-portrait"
                />
                <span>
                  Mobile Portrait
                </span>
              </span>
            </span>
          </dd>
        </dl>
        <span
          className="vcv-ui-navbar-control vcv-ui-pull-end"
          data-vcv-guide-helper="hub-control"
          onClick={[Function]}
          title="Visual Composer Hub"
        >
          <span
            className="vcv-ui-navbar-control-content"
          >
            <i
              className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-hub-shop"
            />
            <span>
              Visual Composer Hub
            </span>
          </span>
        </span>
        <span
          className="vcv-ui-navbar-control vcv-ui-pull-end"
          data-vcv-guide-helper="settings-control"
          onClick={[Function]}
          title="Settings"
        >
          <span
            className="vcv-ui-navbar-control-content"
          >
            <i
              className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-cog"
            />
            <span>
              Settings
            </span>
          </span>
        </span>
        <span
          className="vcv-ui-navbar-control-separator vcv-ui-pull-end"
        />
        <div
          className="vcv-ui-navbar-controls-group vcv-ui-pull-end"
          data-vcv-guide-helper="save-control"
        >
          <span
            className="vcv-ui-navbar-control"
            onClick={[Function]}
          >
            <span
              className="vcv-ui-navbar-control-content"
            >
              <i
                className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-save"
              />
              <span />
            </span>
          </span>
        </div>
        <dl
          className="vcv-ui-navbar-dropdown vcv-ui-pull-end vcv-ui-navbar-sandwich"
        >
          <dt
            className="vcv-ui-navbar-dropdown-trigger vcv-ui-navbar-control"
            onClick={[Function]}
            title="Menu"
          >
            <span
              className="vcv-ui-navbar-control-content"
            >
              <i
                className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-mobile-menu"
              />
              <span>
                Menu
              </span>
            </span>
          </dt>
          <dd
            className="vcv-ui-navbar-dropdown-content vcv-ui-navbar-show-labels"
          >
            <div
              style={
                Object {
                  "height": "auto",
                  "maxHeight": "80vh",
                  "minHeight": 0,
                  "overflow": "hidden",
                  "position": "relative",
                  "width": "100%",
                }
              }
            >
              <div
                className="vcv-ui-scroll-content"
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "bottom": undefined,
                    "left": undefined,
                    "marginBottom": 0,
                    "marginRight": 0,
                    "maxHeight": "calc(80vh + 0px)",
                    "minHeight": 0,
                    "overflow": "scroll",
                    "position": "relative",
                    "right": undefined,
                    "top": undefined,
                  }
                }
              >
                <div>
                  <div
                    className="vcv-ui-navbar-controls-set"
                  >
                    <span
                      className="vcv-ui-navbar-control"
                      onClick={[Function]}
                    >
                      <span
                        className="vcv-ui-navbar-control-content"
                      />
                    </span>
                    <span
                      className="vcv-ui-navbar-control"
                      onClick={[Function]}
                    >
                      <span
                        className="vcv-ui-navbar-control-content"
                      />
                    </span>
                    
                    <span
                      className="vcv-ui-navbar-control"
                      data-backend-editor="backendEditor"
                      onClick={[Function]}
                    >
                      <span
                        className="vcv-ui-navbar-control-content"
                      />
                    </span>
                  </div>
                </div>
              </div>
              <div
                className="vcv-ui-scroll-track--horizontal"
                style={
                  Object {
                    "display": "none",
                    "height": 6,
                    "position": "absolute",
                  }
                }
              >
                <div
                  className="vcv-ui-scroll-thumb--horizontal"
                  style={
                    Object {
                      "display": "block",
                      "height": "100%",
                      "position": "relative",
                    }
                  }
                />
              </div>
              <div
                className="vcv-ui-scroll-track--vertical"
                style={
                  Object {
                    "display": "none",
                    "position": "absolute",
                    "width": 6,
                  }
                }
              >
                <div
                  className="vcv-ui-scroll-thumb--vertical"
                  style={
                    Object {
                      "display": "block",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </dd>
        </dl>
        <div
          className="vcv-ui-navbar-drag-handler vcv-ui-navbar-controls-spacer"
          onMouseDown={[Function]}
        />
      </nav>
    </div>
  </div>
</div>
`;
